controllers/fenceagentsremediation_controller.go

```diff
@@ -74,39 +75,70 @@ func (r *FenceAgentsRemediationReconciler) Reconcile(ctx context.Context, req ct
 	emptyResult := ctrl.Result{}
 
 	// Fetch the FenceAgentsRemediation instance
-	far := &v1alpha1.FenceAgentsRemediation{}
+	far := &farV1alpha1.FenceAgentsRemediation{}
 	if err := r.Get(ctx, req.NamespacedName, far); err != nil {
 		if apiErrors.IsNotFound(err) {
-			// FAR is deleted, stop reconciling
-			r.Log.Info("FAR CR is deleted - nothing to do", "CR Name", req.Name, "CR Namespace", req.Namespace)
+			// Request object of FAR CR was not found, could have been deleted after reconcile request.
+			// Return and don't requeue
+			r.Log.Info("FenceAgentsRemediation CR was not found", "CR Name", req.Name, "CR Namespace", req.Namespace)
 			return emptyResult, nil
 		}
-		r.Log.Error(err, "failed to get FAR CR")
+		r.Log.Info("Failed to get FenceAgentsRemediation CR, requeuing.", "CR Name", req.Name)
 		return emptyResult, err
 	}
+	// Add finalizer when object is created
+	if far.ObjectMeta.DeletionTimestamp.IsZero() {
```

Comment:
IIUC this will also be triggered on change of the CR as well, maybe just check if there is already a finalizer.

Reply:
Which kind of change? IIUC it won't be zero only when the CR was deleted (and it's `DeletionTimestamp` is greater than zero)

Reply:
That is correct, however the DeletionTimeStamp will be equal to zero not only when the CR is created (which is the assumption in the code) but also when the CR is modified (without being deleted).

Reply:
> CR is modified (without being deleted)

And this can happen when the FAR's CR will have a status, right?
Anyway, there is no harm of restricting the if clause 👍🏻

Reply:
> And this can happen when the FAR's CR will have a status, right?
Or are you referring to an update by adding/removing the finalizer?

Reply:
it doesn't matter what's causing an update, CRs can be updated, that's it. An empty deletion timestamp just isn't an indicator for a just created resource.

Reply:
> An empty deletion timestamp just isn't an indicator for a just created resource.

How about empty finalizers?